packages/debugger/src/panels/sources/body.ts

```diff
@@ -100,31 +95,36 @@ export class SourcesBody extends Widget {
     if (this._editorHandler) {
       this._editorHandler.dispose();
     }
-
     const { content, mimeType } = source;
     const editorMimeType =
       mimeType ||
-      this._mimeTypeService.getMimeTypeByFilePath(frame.source?.path ?? '');
+      this._mimeTypeService.getMimeTypeByFilePath(
+        this._currentFrame?.source?.path ?? ''
+      );
 
     this._editor.model.sharedModel.setSource(content);
     this._editor.model.mimeType = editorMimeType;
+    console.log(
+      'this._model.currentFrame?.source?.path',
+      this._currentFrame?.source?.path
+    );
 
     this._editorHandler = new EditorHandler({
       debuggerService: this._debuggerService,
       editorReady: () => Promise.resolve(this._editor.editor),
       getEditor: () => this._editor.editor,
-      path: frame.source?.path ?? '',
+      path: this._currentFrame?.source?.path ?? '',
       src: this._editor.model.sharedModel
     });
 
-    this._model.currentSource = {
-      content,
-      mimeType: editorMimeType,
-      path: displayPath
-    };
-
     requestAnimationFrame(() => {
-      EditorHandler.showCurrentLine(this._editor.editor, frame.line, 'start');
+      if (this._currentFrame) {
+        EditorHandler.showCurrentLine(
+          this._editor.editor,
+          this._currentFrame.line,
```

Comment:
I'd introduce a `model.currentLine` on top of to `currentSource`.

```suggestion
          this._currentLine,
```